src/core/database/database.service.ts

```diff
@@ -557,32 +552,45 @@ export class DatabaseService {
 
   async deleteNodeNew<TObject extends Resource>({
     object,
-    baseNodeLabels,
-    uniqueProperties = {},
   }: {
     object: TObject;
-    baseNodeLabels: string[];
-    uniqueProperties?: UniqueProperties<TObject>;
   }) {
     const query = this.db
       .query()
-      .match(node('node', { id: object.id }))
-      //Mark any parent base node relationships (pointing to the base node) as active = false.
+      .match([
+        node('baseNode', { id: object.id }),
+        // in this case we want to set Deleted_ labels for all properties
+        // including those with active = false
+        relation('out', 'propRel'),
+        node('propertyNode', 'Property'),
+      ])
+      // Mark any parent base node relationships (pointing to the base node) as active = false.
       .optionalMatch([
-        node('node'),
+        node('baseNode'),
         relation('in', 'rel'),
         node('', 'BaseNode'),
       ])
-      .set({
-        values: {
-          'rel.active': false,
-        },
+      .setValues({
+        'rel.active': false,
+        'propRel.active': false,
       })
-      .with('distinct(node) as node')
-      //Mark baseNode labels and id deleted
-      .call(setBaseNodeLabelsAndIdDeleted, baseNodeLabels)
-      //Mark unique property labels and values deleted
-      .call(setPropLabelsAndValuesDeleted, uniqueProperties)
+      // after setting propRel.active false, we need to distinct propertyNode (not sure why)
+      .with('baseNode, collect(distinct propertyNode) as propertyNodes')
+      // combine baseNode with propertyNodes to set Deleted_ labels on all of them
+      .with(
+        'reduce(nodes = [baseNode], pn in propertyNodes | nodes + pn) as nodeList'
+      )
```

Comment:
I believe you can just concat the lists together with `+`
```suggestion
      // combine baseNode with propertyNodes to set Deleted_ labels on all of them
      .with('[baseNode] + collect(distinct propertyNode) as nodeList')
```

Reply:
I think that will yield `[baseNode, [[propertyNode, propertyNode, ...]]]`

Reply:
It needs to be flattened

Reply:
Or maybe not. Hmm. Let me see

Reply:
You sure? I don't think so.

Reply:
https://neo4j.com/docs/cypher-manual/current/syntax/operators/#syntax-concatenating-two-lists

Reply:
Ok. You were right.